crypto/src/mls/client/mod.rs

```diff
@@ -192,12 +213,17 @@ impl Client {
 
         let identities = stored_skp.iter().zip(ciphersuites);
 
-        let mut client = Self {
+        let client = ClientInner {
             id: client_id.clone(),
             identities: ClientIdentities::new(stored_skp.len()),
             keypackage_lifetime: KEYPACKAGE_DEFAULT_LIFETIME,
         };
 
+        let mut inner_lock = self.state.lock().await;
+        *inner_lock = Some(client);
+        // Drop the lock because it will be acquired below when the identities are saved.
+        drop(inner_lock);
+
```

Comment:
You use this pattern several times. We can automate it a bit:

```rust
async fn replace_inner(&self, new_inner: ClientInner) {
    let mut inner_lock = self.state.lock().await;
    *inner_lock = Some(new_inner);
}
```

Which lets us simplify the usage:

```rust
self.replace_inner(ClientInner{
    id: client_id.clone(),
    identities: ClientIdentities::new(stores_skp.len()),
    keypackage_lifetime: KEYPACKAGE_DEFAULT_LIFETIME,
}).await;
```